### PR TITLE
Keep the original/sender's casing of our nick on highlight in message view

### PR DESCRIPTION
**Pull request by kripton**
Don't replace the search term by the nick as we know it but keep the message as the sender wrote it

### CODE REVIEW

**Comment by mweibel** [comment]
hm, why? ;)

**Comment by kripton** [comment]
In my opinion it matters if the sender wrote "@mweibel: Hello" or "@MWEIBEL: HELLO". Without using the plugin one would see the difference. With the plugin, the nickname is displayed as the targeted user entered it (or is registered with the server)

**Comment by mweibel** [comment]
@troymccabe @maennchen @benlangfeld @melissanoelle any comments on this? should this be merged or not?

**Comment by benlangfeld** [comment]
I don't have a strong preference either way.

**Comment by bklang** [comment]
I'm not sure if I agree. I'd prefer we normalize to whatever the user entered for a nickname. We want `@benklang` to result in `@BenKlang`, for example.  I'm not sure I'm convinced by the argument that I want to be able to shout at people by capitalizing their nicknames.

**Comment by maennchen** [comment]
I think @bklang is right. We should continue to normalize the usernames.

**Comment by maennchen** [comment]
If this is an actual need for someone we should maybe implement a config switch to choose if it should be normalized.

**Comment by kripton** [comment]
It's not a real need, I just thought it was a glitch since I'm not used to "chat clients" modifying the message (even if it's only the casing) compared to what was received.
Feel free to close/reject this PR if the general agreement is to continue normalizing the nicknames.

**Comment by kripton** [comment]
Just changed my mind: I'll make it configurable this evening and update this PR then.

**Comment by kripton** [comment]
PR updated. Now the behaviour is controlled by a configuration option.

File: notifyme/candy.js [hunk 1]
```diff
@@ -77,7 +79,11 @@ CandyShop.NotifyMe = (function(self, Candy, $) {
 			if (searchRegExp.test(args.templateData.message) && args.templateData.name != _getNick()) {
 				// highlight if specified
 				if (_options.highlightInRoom) {
-					args.templateData.message = args.templateData.message.replace(searchRegExp, '$1<span class="candy-notifyme-highlight">' + searchTerm + '</span>');
+					if (_options.normalizeNickname) {
+						args.templateData.message = args.templateData.message.replace(searchRegExp, '$1<span class="candy-notifyme-highlight">' + searchTerm + '</span>');
```
Comment (benlangfeld) [inline notifyme/candy.js]: It would be nice to remove the duplication here and leave only the content of the tag inside the conditional.

**Comment by kripton** [comment]
Updated PR to address comment of benlangfeld

**Comment by mweibel** [comment]
thanks :+1: will be granting you push access to this repo. We have the rule that updating your own plugins is ok without PRs (even though I still prefer having PRs for that), updating other contributors' plugins needs a PR.

**Pull request merged**